Sources/Apodini/Components/Component.swift

```diff
@@ -7,37 +7,84 @@
 
 
 /// A `Component` is the central building block of  Apodini. Each component handles a specific functionality of the Apodini web service.
 ///
-/// A `Component` either has a `handle` function that is called when a request reaches the `Component` or consists of different other components as described by the `content` property.
+/// A `Component`  consists of different other components as described by the `content` property.
 public protocol Component {
     /// The type of `Component` this `Component` is made out of if the component is a composition of multiple subcomponents.
-    associatedtype Content: Component = Never
-    /// The type that is returned from the `handle` method when the component handles a request. The return type of the `handle` method is encoded into the response send out to the client.
-    associatedtype Response: Encodable = Never
-    
+    associatedtype Content: Component
     
     /// Different other `Component`s that are composed to describe the functionality of the`Component`
-    @ComponentBuilder var content: Self.Content { get }
+    @ComponentBuilder
+    var content: Content { get }
+}
+
+
+/// A `Handler` is a `Component` which defines an endpoint and can handle requests.
+public protocol Handler: Component {
+    /// The type that is returned from the `handle()` method when the component handles a request. The return type of the `handle` method is encoded into the response send out to the client.
+    associatedtype Response: Encodable
     
+    /// A function that is called when a request reaches the `Handler`
+    func handle() -> Response
+}
+
+
+extension Handler {
+    public var content: some Component { EmptyComponent() }
+}
+
+
+/// A `Handler` which can be uniquely identified
+public protocol IdentifiableHandler: Handler {
+    /// The type of this handler's identifier
+    associatedtype EndpointIdentifier: AnyEndpointIdentifier
     
-    /// A function that is called when a request reaches the `Component`
-    func handle() -> Self.Response
+    /// This handler's identifier
+    var endpointId: EndpointIdentifier { get }
 }
 
 
 extension Component {
     func visit(_ visitor: SyntaxTreeVisitor) {
-        precondition(((try? typeInfo(of: Self.self).kind) ?? .none) == .struct, "Component \((try? typeInfo(of: Self.self).name) ?? "unknown") must be a struct")
-        
+        LKAssertTypeIsStruct(Self.self)
         if let visitable = self as? Visitable {
             visitable.visit(visitor)
-        } else if Self.Content.self != Never.self {
-            content.visit(visitor)
         } else {
-            visitor.register(component: self)
+            if Content.self != Never.self {
+                visitor.enterCollection()
+                content.visit(visitor)
+                visitor.exitCollection()
+            }
+            HandlerVisitorHelperImpl(visitor: visitor)(self)
         }
     }
 }
+
+
+private func LKAssertTypeIsStruct<T>(_: T.Type) {
```

Comment:
I would suggest to remove the `LK` prefix.
```suggestion
private func AssertTypeIsStruct<T>(_: T.Type) {
```

Reply:
good catch i missed that one